src/jobflow_remote/jobs/runner.py

```diff
@@ -275,15 +275,9 @@ def run(
                 self.checkout
             )
 
-        if transfer or queue or complete:
-            try:
-                self.advance_state(states)
-            except Exception:
-                logger.exception("Error during initial advance_state")
-            scheduler.every(self.runner_options.delay_advance_status).seconds.do(
-                self.advance_state, states=states
-            )
-
+        # first perform this step with check run status and update the status of
+        # the limited workers (if any). Otherwise, upon start of the runner the initial
+        # count of the currently submitted jobs is 0.
```

Comment:
Any way to test that somehow ? How did you find out about that bug ?

Reply:
I stopped and restarted the runner and more jobs were submitted, even if the maximum was already reached. There is already a test for the workers with maximum number of jobs: https://github.com/Matgenix/jobflow-remote/blob/a68ff9d2cae0d91478b85c440ddaf4f44742af86/tests/integration/test_advanced_options.py#L81
The problem is that it is already a bit tricky and I would need to add to this the fact of stopping and restarting the runner. Plus I would need to make the jobs sleep for at least 30 seconds to make sure that they don't finish during the restart of the runner. Still, it should be feasible, if you think it is better I can try to add it.

Reply:
I don't know, it's just that if for some reason, we forget about this, and/or someone else modifies the code (e.g. moving the snippets of code for some reason) and we approve (because we forgot), then we get it back. I would say it would probably be a good thing to have.